metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala

```diff
@@ -48,25 +48,27 @@ final class DefinitionProvider(
       params: TextDocumentPositionParams,
       token: CancelToken
   ): DefinitionResult = {
-    semanticdbs.textDocument(path).documentIncludingStale match {
-      case Some(doc) =>
-        definitionFromSnapshot(path, params, doc)
-      case _ =>
-        val fromCompilers =
-          compilers()
-            .definition(params, token)
-            .getOrElse(Collections.emptyList())
-        if (!fromCompilers.isEmpty()) {
-          DefinitionResult(
-            fromCompilers,
-            "",
-            None,
-            None
-          )
-        } else {
+    val fromSemanticdb =
+      semanticdbs.textDocument(path).documentIncludingStale match {
+        case Some(doc) =>
+          definitionFromSnapshot(path, params, doc)
+        case _ =>
           warnings.noSemanticdb(path)
```

Comment:
does it make sense to warn even in cases in which there's a fallback available?
Mayne the warning could include info in whether Metals has found the fallback or not

Reply:
I decided to keep the warning and leave it to a separate PR to overall improve the UX (reduce warnings/noise) when semanticdb-scalac is not enabled.

Reply:
This warning only happens when using "rambo mode" (editing a source file outside a build target).

Reply:
We don't have the classpath for those source files so I suspect definition/completions won't work very reliably.

Reply:
Got it, makes sense.
Incidentally local completions now work in project/*.scala files, which was an unexpected treat 😅

Reply:
After this PR goto definition will also work there for local variables :)